plugins/couchbase/src/main/java/org.radargun.service/CouchbaseOperations.java

```diff
@@ -0,0 +1,70 @@
+package org.radargun.service;
+
+import com.couchbase.client.java.document.ByteArrayDocument;
+import org.radargun.traits.BasicOperations;
+
+import java.nio.charset.StandardCharsets;
+
+public class CouchbaseOperations implements BasicOperations {
+
+   protected final CouchbaseService service;
+   private final CouchbaseCacheAdapter adapter = new CouchbaseCacheAdapter();
+
+   public CouchbaseOperations(CouchbaseService service) {
+      this.service = service;
+   }
+
+   @Override
+   public <K, V> Cache<K, V> getCache(String cacheName) {
+      return adapter;
+   }
+
+   private class CouchbaseCacheAdapter<K, V> implements BasicOperations.Cache<byte[], byte[]> {
+
+      @Override
+      public byte[] get(byte[] key) {
+         ByteArrayDocument document = service.bucket.get(new String(key, StandardCharsets.UTF_8), ByteArrayDocument.class);
+         return document.content();
+      }
+
+      @Override
+      public boolean containsKey(byte[] key) {
+         return service.bucket.exists(new String(key));
+      }
+
+      @Override
+      public void put(byte[] key, byte[] value) {
+         ByteArrayDocument doc = ByteArrayDocument.create(new String(key), value);
+         service.bucket.upsert(doc);
+      }
+
+      @Override
+      public byte[] getAndPut(byte[] key, byte[] value) {
+         byte[] document = get(key);
+         put(key, value);
+         return document;
+      }
+
+      @Override
+      public boolean remove(byte[] key) {
+         try {
+            service.bucket.remove(new String(key));
+            return true;
+         } catch(Exception e) {
+            return false;
+         }
+      }
+
+      @Override
+      public byte[] getAndRemove(byte[] key) {
+         byte[] doc = get(key);
```

Comment:
I think this should use the cas on remove to be more realistic.

Might make sense to have specialized impls for this to avoid creating a string key more than once

Reply:
Sure, will look to do that.